offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/OfferToPDFConverter.groovy

```diff
@@ -479,78 +481,110 @@ class OfferToPDFConverter implements OfferExporter {
 
         }
 
-        static String tableHeader(String elementId) {
-            //1. add pagebreak
-            //2. create empty table for elementId
+        static String pageBreak() {
+
             return """<div class="pagebreak"></div>
-                                     <div class="row table-header" id="grid-table-header">
-                                         <div class="col-1">Pos.</div>
+                   """
+        }
+
+        static String tableHeader() {
+
+            return """<div class="row table-header" id="grid-table-header-${tableCount}">
+                                         <div class="col-1">&#8470;</div>
                                          <div class="col-4">Service Description</div>
                                          <div class="col-1 price-value">Amount</div>
                                          <div class="col-2 text-center">Unit</div>
                                          <div class="col-2 price-value">Price/Unit (€)</div>
                                          <div class="col-2 price-value">Total (€)</div>
                                     </div>
-                                 <div class="product-items" id="${elementId}"></div>
-                             """
+                                    """
+        }
+
+        static String createNewTable(String elementId) {
+            """<div class="product-items" id="${elementId}"></div>
+            """
```

Comment:
do you think its possible here to refactor the method so that it starts the new product table in a way that it also creates the title and the header? This always needs to be set for a new table, right?

Reply:
That's a good question. Sadly we can't include the title since we also have the case that a productcategory has more productItems than can be displayed on one page. 
In this case we create a new table storing the remaining productItems on the following page(s) without repeating the title. I'll check if this method can be merged with the header generation method though 👍

Reply:
Sadly this doesn't work since we have multiple cases in which a new table is necessary and they vary in the necessity and ordering of: 

1. page break 
2. title
3. header